tree-sitter-dscexpression/grammar.js

```diff
@@ -18,19 +18,20 @@ module.exports = grammar({
     escapedStringLiteral: $ => token(prec(PREC.ESCAPEDSTRING, seq('[[', /.*?/))),
     bracketInStringLiteral: $ => token(prec(PREC.BRACKETINSTRING, seq('[', /.*?/, ']', /.+?/))),
     _expressionString: $ => prec(PREC.EXPRESSIONSTRING, seq('[', $.expression, ']')),
-    expression: $ => seq($.function, optional($._members)),
+    expression: $ => seq(field('function', $.function), field('members', optional($.memberAccess))),
     stringLiteral: $ => token(prec(PREC.STRINGLITERAL, /[^\[].*?/)),
 
-    function: $ => seq($.functionName, '(', optional($._arguments), ')'),
-    functionName: $ => /[a-zA-Z]+/,
-    _arguments: $ => seq($._argument, repeat(seq(',', $._argument))),
-    _argument: $ => choice($.expression, $.string, $.number, $.boolean),
+    function: $ => seq(field('name', $.functionName), '(', field('args', optional($.arguments)), ')'),
+    functionName: $ => /[a-z][a-zA-Z0-9]*/,
+    arguments: $ => seq($._argument, repeat(seq(',', $._argument))),
+    _argument: $ => choice($.expression, seq('\'', $.string, '\''), $.number, $.boolean),
 
-    string: $ => seq("'", /[^']*/, "'"),
-    number: $ => /\d+/,
+    // ARM strings do not allow to contain single-quote characters
+    string: $ => /[^']*/,
+    number: $ => /-?\d+/,
     boolean: $ => choice('true', 'false'),
 
-    _members: $ => repeat1($._member),
+    memberAccess: $ => repeat1($._member),
```

Comment:
Let's discuss this at the review

Reply:
I still want the `memberAccess` parent node over the actual `memberName` nodes, so changing the grammar so `memberAccess` is not optional and the definition of `memberAccess` is to `repeat(member)` doesn't work as tree-sitter complains that this can match an empty string.